cmd/flexctl/backups.go

```diff
@@ -224,3 +227,126 @@ func listBackups(cmd *cobra.Command) error {
 func backupsEnabled() bool {
 	return os.Getenv("S3_ARCHIVE_CONFIG") != ""
 }
+
+func newBackupConfigCmd() *cobra.Command {
+	var configCmd = &cobra.Command{
+		Use:   "config",
+		Short: "Manage backup configuration",
+	}
+
+	configCmd.AddCommand(newConfigShowCmd())
+	configCmd.AddCommand(newConfigUpdateCmd())
+
+	return configCmd
+}
+
+type configShowResult struct {
+	Result flypg.BarmanSettings `json:"result"`
+}
+
+func newConfigShowCmd() *cobra.Command {
+	var configShowCmd = &cobra.Command{
+		Use:   "show",
+		Short: "Show current configuration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			resp, err := http.Get("http://localhost:5500/commands/admin/settings/view/barman")
+			if err != nil {
+				return err
+			}
+
+			var rv configShowResult
+			err = json.NewDecoder(resp.Body).Decode(&rv)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("  ArchiveTimeout = %s\n", rv.Result.ArchiveTimeout)
+			fmt.Printf("  RecoveryWindow = %s\n", rv.Result.RecoveryWindow)
+			fmt.Printf("  FullBackupFrequency = %s\n", rv.Result.FullBackupFrequency)
+			fmt.Printf("  MinimumRedundancy = %s\n", rv.Result.MinimumRedundancy)
+
+			return nil
+		},
+	}
+
+	return configShowCmd
+}
+
+type successfulUpdateResult struct {
+	Message         string `json:"message,omitempty"`
+	RestartRequired bool   `json:"restart_required"`
+}
+
+type configUpdateResult struct {
+	Result successfulUpdateResult `json:"result,omitempty"`
+	Error  string                 `json:"error,omitempty"`
+}
+
+func newConfigUpdateCmd() *cobra.Command {
+	var configUpdateCmd = &cobra.Command{
+		Use:   "update",
+		Short: "Update configuration",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			update := flypg.BarmanSettings{
+				ArchiveTimeout:      cmd.Flag("archive-timeout").Value.String(),
+				RecoveryWindow:      cmd.Flag("recovery-window").Value.String(),
+				FullBackupFrequency: cmd.Flag("full-backup-frequency").Value.String(),
+				MinimumRedundancy:   cmd.Flag("minimum-redundancy").Value.String(),
+			}
+
+			jsonBody, err := json.Marshal(update)
+			if err != nil {
+				return err
+			}
+
+			resp, err := http.Post("http://localhost:5500/commands/admin/settings/update/barman", "application/json", bytes.NewBuffer(jsonBody))
```

Comment:
Instead of localhost, let's use `$FLY_APP_NAME.internal`.  This should insure that the request hits the primary in the event this is called from a replica.

Reply:
I thought about using `$FLY_APP_NAME` for this, but then it'd tie postgres-flex to Fly, which is something I thought we were trying to avoid. Agreed that it's a problem though--I only tested from what was likely the primary. Thoughts on a platform-agnostic fix? Or is platform-agnosticity not the priority I read it as?

Reply:
There's no real intention to make this project platform agnostic.

Reply:
You may be referring to my previous point about not tying ourselves to a specific s3 vendor?

Reply:
Ah, yeah, that was it. I over-extrapolated and incorrectly assumed we were building a more generic project that just happened to run better on fly.io. Thanks for clarifying, I'll switch all the URLs over tomorrow.